src/main/java/com/ericsson/eiffelcommons/jenkins/JenkinsManager.java

```diff
@@ -43,56 +43,46 @@
     /**
      * Constructor, takes jenkins baseUrl, username and password.
      *
-     * @param jenkinsBaseUrl
-     *            :: Jenkins base url http://somehost:123
-     * @param username
-     *            :: Jenkins username as String
-     * @param password
-     *            :: Jenkins password or API-token
+     * @param jenkinsBaseUrl :: Jenkins base url http://somehost:123
+     * @param username       :: Jenkins username as String
+     * @param password       :: Jenkins password or API-token
      * @throws URISyntaxException
      * @throws JSONException
      * @throws IOException
      * @throws ClientProtocolException
      */
     public JenkinsManager(String jenkinsBaseUrl, String username, String password)
-            throws URISyntaxException, JSONException, ClientProtocolException, IOException {
+            throws ClientProtocolException, URISyntaxException, IOException {
         this.jenkinsBaseUrl = jenkinsBaseUrl;
         this.encoding = createEncodingFromUsernameAndPassword(username, password);
-        this.crumb = getCrumb();
+        this.crumb = fetchCrumbFromJenkins();
```

Comment:
Instead of duplicating code, couldn't this constructor call on the other one passing default values for the unknown parameters? Also I think the javadoc should be extended to include information about what a crumb is and why it's needed...

Reply:
One constructor now calls the other. I also added some more javadoc to the fetchCrumbFromJenkins method.